### ISSUE TITLE
Compare the performance 

### ISSUE THREAD

**Issue by shivahanifi** [shivahanifi/SCDD-image-segmentation-keras#25]
The performance on the UNet model is reported in terms of precision and recall for 5 classes in the thesis. I trained the model for 65 epochs imposing the class weights recommended by the authors (class C #2 ). The following demonstrates the comparison between two models.

||reported|trained|
|-|-|-|
|precision|![Screenshot from 2024-11-19 11-42-38](https://github.com/user-attachments/assets/0ac1dad6-ac18-4e8a-9698-6b7357c0dea1)|![Screenshot from 2024-11-19 10-00-17](https://github.com/user-attachments/assets/9cbdaa37-f3fc-4bfb-a39b-e6f4c08ce242)|
|recall|![Screenshot from 2024-11-19 11-42-50](https://github.com/user-attachments/assets/b8ed82ff-0460-4b41-b404-56ea00193eb1)|![Screenshot from 2024-11-19 10-00-04](https://github.com/user-attachments/assets/e0d9d9b2-df35-4020-8a24-a9aa9ad5dbe7)|

### Interpretation

**diagonal elements** 
represent the recall and precision for each class
- the precision for the background is 0.97, meaning 97% of the instances predicted as background are correctly identified as background.
- recall is 0.94, meaning 94% of the actual background instances were identified correctly.
- since both precision and recall are high for the backgtround class, means the model is effective at identifying background instances, both in terms of correctly predicting them and identifying most of the background labels.

**off-diagonal elements**
represent misclassifications.
- having higher values on the row corresponding to the background class in the precision matrix means that while the true label is background, for example 

The correlation of precision and recall in this confusion matrix gives you a sense of how balanced or biased the model is in its predictions. A perfect model would have both high precision and high recall across all classes, which would result in all diagonal elements being close to 1.

## Detailed comparison
Higher precision means less false positives and higher recall means betetr at identifying all the instances of the class.
|class|reported precision|trained precision||reported recall|trained recall|
|-|-|-|-|-|-|
|background|1.00|0.97||0.92|0.94|
|sp multi|0.85|0.86||0.98|0.83|
|ribbons|0.44|0.68||1.00|0.86|
|gridlines|0.37|0.42||0.82|0.60|
|crack|0.18|0.33||1.00|0.74|

While the trained model is doing well in avoiding false positives (better or similar precision), it is struggling to identify all relevant instances of the target classes (lower recall). Specifically "ribbons" and "creack" show the most notable drop in recall.
